cwe_checker_rs/src/analysis/graph.rs

```diff
@@ -311,6 +350,7 @@ impl<'a> GraphBuilder<'a> {
         for (call_node, return_to_node) in self.return_addresses[&return_from_sub.tid].iter() {
             let (call_block, caller_sub) = match self.graph[*call_node] {
                 Node::BlkEnd(block, sub) => (block, sub),
+                Node::CallSource { source, .. } => source,
```

Comment:
As mentioned, the graph should now disallow `CallReturn` nodes without corresponding `CallSource` nodes. So the `Node::BlkEnd` case should panic.